app/src/main/java/com/mealsmadeeasy/ui/home/profile/ProfileFragment.kt

```diff
@@ -0,0 +1,148 @@
+
+        // Populate sex spinner values
+        val sexSpinner = root.findViewById<Spinner>(R.id.sex_spinner)
+        var sexValues: List<String> = emptyList()
+        for (item in Gender.values()) {
+            sexValues += item.toString()
+        }
+        sexValues += getString(R.string.profile_spinner_prompt)
+        val sexAdapter = HintAdapter(this.activity, android.R.layout.simple_list_item_1, sexValues)
+        sexAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        sexSpinner.adapter = sexAdapter
+        sexSpinner.setSelection(sexAdapter.count)
+
+        // Set up birthday date picker
+        val dateText = root.findViewById<EditText>(R.id.birthday_edit_text)
+        val cal = Calendar.getInstance()
+        setupDatePicker(dateText, cal)
+
+        // Set up text input view hints
+        val feetContainer = root.findViewById<TextInputLayout>(R.id.height_feet_container)
+        val feetEditText = root.findViewById<TextInputEditText>(R.id.height_feet_edit_text)
+        val inchesContainer = root.findViewById<TextInputLayout>(R.id.height_inches_container)
+        val inchesEditText = root.findViewById<TextInputEditText>(R.id.height_inches_edit_text)
+        val weightContainer = root.findViewById<TextInputLayout>(R.id.weight_container)
+        val weightEditText = root.findViewById<TextInputEditText>(R.id.weight_field)
+        feetEditText.onFocusChangeListener = TextInputViewHintListener(feetContainer,
+                getString(R.string.profile_height_prompt), getString(R.string.profile_height_feet_units))
+        inchesEditText.onFocusChangeListener = TextInputViewHintListener(inchesContainer,
+                getString(R.string.profile_height_inches_prompt), getString(R.string.profile_height_inches_units))
+        weightEditText.onFocusChangeListener = TextInputViewHintListener(weightContainer,
+                getString(R.string.profile_weight_prompt), getString(R.string.profile_weight_lbs_units))
+
+        // Set save button behavior
+        val saveButton = root.findViewById<Button>(R.id.save_button)
+        saveButton.setOnClickListener{
+            val sex = sexSpinner.selectedItem
+            val age = cal.timeInMillis
+            val feet = feetEditText.text.toString()
+            val inches = inchesEditText.text.toString()
+            val pounds = weightEditText.text.toString()
+
+            val builder = AlertDialog.Builder(this.activity)
+            if (sex == getString(R.string.profile_spinner_prompt)
+                    || dateText.text.toString() == ""
+                    || feet == "" || inches == "" || pounds == "") {
+                builder.setMessage(R.string.error_profile_blank_field)
+            } else if (age.toInt() < MIN_AGE) {
+                builder.setMessage(R.string.error_profile_underage)
+            } else if (inches.toInt() >= INCHES_IN_FEET) {
+                builder.setMessage(R.string.error_profile_invalid_height_inches)
```

Comment:
This a total pain to do, so I'm fine if you decide not to go this route, but because these are all client-side checks, it's normally easier for users if you show show errors under fields as they type. It does make sense for the BMI check to happen in an alert, though.

Reply:
But errors appearing as I'm not finished typing my input irritate me.

Reply:
I mean, if you want to make it even more of a pain, you could also wait until the user moves to another field before showing an error.